benchmarks/distributed/rpc/ps/metrics/CUDAMetric.py

```diff
@@ -0,0 +1,25 @@
+import torch
+
+from .MetricBase import MetricBase
+
+
+class CUDAMetric(MetricBase):
+    def __init__(self, name: str):
+        self.name = name
+        self.start = None
+        self.end = None
+
+    def record_start(self, rank):
+        self.start = torch.cuda.Event(enable_timing=True)
+        with torch.cuda.device(rank):
+            self.start.record()
+
+    def record_end(self, rank):
+        self.end = torch.cuda.Event(enable_timing=True)
+        with torch.cuda.device(rank):
+            self.end.record()
+
+    def elapsed_time(self, rank):
+        self.start.synchronize()
+        self.end.synchronize()
+        return self.start.elapsed_time(self.end)
```

Comment:
IIRC, this will fail if `self.start` and `self.end` are on different devices?

Reply:
Yes, it does fail. The API calls to `record_start` and `record_end` are from the logger. The logger is initialized with a device rank. So it shouldn't be a problem if a user is inheriting from a base class since they can't access the logger because of name mangling.